spec/cedar_policy/authorizer_spec.rb

```diff
@@ -172,5 +172,152 @@
+          [
+            {uid: {type: "User", id: "1"}, attrs: {}, parents: []},
+            {uid: {type: "Image", id: "1"}, attrs: {owner: {type: "User", id: "1"}}, parents: []}
+          ]
+        }
+
+        it { is_expected.to(have_attributes(decision: CedarPolicy::Decision::ALLOW)) }
+      end
+
+      describe "Non-Owner cannot Edit image" do
+        let(:principal) { CedarPolicy::EntityUid.new("User", "3") }
+        let(:action) { CedarPolicy::EntityUid.new("Action", "Edit") }
+        let(:entities_data) {
+          [
+            {uid: {type: "User", id: "1"}, attrs: {}, parents: []},
+            {uid: {type: "User", id: "3"}, attrs: {}, parents: []},
+            {uid: {type: "Image", id: "1"}, attrs: {owner: {type: "User", id: "1"}}, parents: []}
+          ]
+        }
+
+        it { is_expected.to(have_attributes(decision: CedarPolicy::Decision::DENY)) }
+      end
+
+      describe "Admin can Delete image" do
+        let(:principal) { CedarPolicy::EntityUid.new("User", "2") }
+        let(:action) { CedarPolicy::EntityUid.new("Action", "Delete") }
+        let(:entities_data) {
+          [
+            {uid: {type: "User", id: "1"}, attrs: {}, parents: []},
+            {uid: {type: "User", id: "2"}, attrs: {isAdmin: true}, parents: []},
+            {uid: {type: "Image", id: "1"}, attrs: {owner: {type: "User", id: "1"}}, parents: []}
+          ]
+        }
+
+        it { is_expected.to(have_attributes(decision: CedarPolicy::Decision::ALLOW)) }
+      end
+
+      describe "entity with superfluous attribute" do
+        let(:entities_data) {
+          [
+            {uid: {type: "User", id: "1"}, attrs: {extra: "0xfeedc0de"}, parents: []},
+            {uid: {type: "User", id: "2"}, attrs: {isAdmin: true}, parents: []},
+            {uid: {type: "Image", id: "1"}, attrs: {owner: {type: "User", id: "1"}}, parents: []}
+          ]
+        }
+
+        it { expect { authorizer.authorize(request, policy_set, entities) }.to(raise_error(ArgumentError)) }
+      end
+
+      describe "entity with missing attribute" do
+        let(:entities_data) {
+          [
+            {uid: {type: "User", id: "1"}, attrs: {}, parents: []},
+            {uid: {type: "User", id: "2"}, attrs: {isAdmin: true}, parents: []},
+            {uid: {type: "Image", id: "1"}, attrs: {}, parents: []}
+          ]
+        }
+
+        it { expect { authorizer.authorize(request, policy_set, entities) }.to(raise_error(ArgumentError)) }
+      end
+
+      describe "entity with incorrectly-typed attribute" do
+        let(:entities_data) {
+          [
+            {uid: {type: "User", id: "1"}, attrs: {}, parents: []},
+            {uid: {type: "User", id: "2"}, attrs: {isAdmin: "0xfeedc0de"}, parents: []},
+            {uid: {type: "Image", id: "1"}, attrs: {owner: {type: "User", id: "1"}}, parents: []}
+          ]
+        }
+
+        it { expect { authorizer.authorize(request, policy_set, entities) }.to(raise_error(ArgumentError)) }
```

Comment:
Prefer to use `raise_error(ArgumentError, /.../)` to explain which error message is returned